### ISSUE TITLE
public docker image

### ISSUE THREAD

**Issue by matti** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#155]
please publish the images - for example this https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/master/docker_native_scripts/Dockerfile takes a long time to build

**Comment by zhiyua-git** [comment]
Hi @matti ,

The images are already published. You can retrieve them following https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/examples-gstreamer-plugin.html#examples-gstreamer-plugin-docker-download.

Hope this helps.
-ZH

**Comment by matti** [comment]
they are not public and it seems like this one https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/master/docker_native_scripts/Dockerfile is not published

**Comment by zhiyua-git** [comment]
You need to follow the authentication steps first in the document: https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/examples-gstreamer-plugin.html#examples-gstreamer-plugin-docker-authenticate.

The docker file you referred is built and published as Ubuntu image: 
Download the Docker Image for Ubuntu
sudo docker pull 546150905175.dkr.ecr.us-west-2.amazonaws.com/kinesis-video-producer-sdk-cpp-amazon-linux:latest

**Comment by matti** [comment]
yeah, I did see the authentication - would be more convenient to have fully public images.
